### PR TITLE
Avoid exception in case a membership is already found.

### PR DESCRIPTION
**Pull request by idimopoulos**
From issue https://github.com/Gizra/og/issues/295

### CODE REVIEW

File: og.module [hunk 1]
```diff
@@ -47,8 +47,11 @@ function og_entity_insert(EntityInterface $entity) {
     return;
   }
 
-  $membership = Og::createMembership($entity, $entity->getOwner());
-  $membership->save();
+  // Other modules might interfere and create a membership.
+  if (!Og::getMembership($entity, $entity->getOwner())) {
```
Comment (amitaibu) [inline og.module]: I think here you will get only the `active` memberships.

File: og.module [hunk 2]
```diff
@@ -47,8 +47,11 @@ function og_entity_insert(EntityInterface $entity) {
     return;
   }
 
-  $membership = Og::createMembership($entity, $entity->getOwner());
-  $membership->save();
+  // Other modules might interfere and create a membership.
```
Comment (amitaibu) [inline og.module]: maybe `// Implementing modules might have created a membership ahead of us.`

**Review by amitaibu — COMMENTED** [review]
Thanks. I'd love to get a test as-well to confirm the functionality.

File: og.module (hunk 1 above)
Comment (pfrenssen) [reply og.module]: Indeed!! I have also been trapped by this in the past, which is why I created #292 to lobby for changing this to return all memberships by default. I think it is not intuitive that this method returns filtered results by default, and an additional parameter should be passed to get the actual membership.

**Comment by pfrenssen** [comment]
I think it's OK for me to merge this since I did not write the original patch but just the test coverage that proves the bug is real.

**Pull request merged**